Tasks/WindowsMachineFileCopy/WindowsMachineFileCopy.ps1

```diff
@@ -1,122 +1,104 @@
-$sourcePath = $sourcePath.Trim('"')
-$targetPath = $targetPath.Trim('"')
+Trace-VstsEnteringInvocation $MyInvocation
 
-# Default + constants #
-$resourceFQDNKeyName = Get-ResourceFQDNTagKey
+# Get inputs for the task
+$machineNames = Get-VstsInput -Name MachineNames
+$adminUserName = Get-VstsInput -Name AdminUserName
+$adminPassword = Get-VstsInput -Name AdminPassword
+$sourcePath = Get-VstsInput -Name SourcePath
+$targetPath = Get-VstsInput -Name TargetPath
+$additionalArguments = Get-VstsInput -Name AdditionalArguments
+$cleanTargetBeforeCopy = Get-VstsInput -Name CleanTargetBeforeCopy
+$copyFilesInParallel = Get-VstsInput -Name CopyFilesInParallel
 
-$envOperationStatus = 'Passed'
+# Import the loc strings.
+Import-VstsLocStrings -LiteralPath $PSScriptRoot/Task.json
 
-Validate-SourcePath $sourcePath
-Validate-DestinationPath $targetPath $environmentName
+. $PSScriptRoot/RoboCopyJob.ps1
+. $PSScriptRoot/Utility.ps1
 
-if([string]::IsNullOrWhiteSpace($environmentName))
+try 
 {
-    Write-Verbose "No environment found. Copying to destination."
+    # keep machineNames parameter name unchanged due to back compatibility
+    $sourcePath = $sourcePath.Trim('"')
+    $targetPath = $targetPath.Trim('"')
 
-    Write-Output (Get-LocalizedString -Key "Copy started for - '{0}'" -ArgumentList $targetPath)
-    Copy-OnLocalMachine -sourcePath $sourcePath -targetPath $targetPath -adminUserName $adminUserName -adminPassword $adminPassword `
-                        -cleanTargetBeforeCopy $cleanTargetBeforeCopy -additionalArguments $additionalArguments
-    Write-Verbose "Files copied to destination successfully."
-}
-else
-{
+    $envOperationStatus = 'Passed'
 
-    $connection = Get-VssConnection -TaskContext $distributedTaskContext
+    Validate-SourcePath $sourcePath
+    Validate-DestinationPath $targetPath $machineNames
 
-    Write-Verbose "Starting Register-Environment cmdlet call for environment : $environmentName with filter $machineFilter"
-    $environment = Register-Environment -EnvironmentName $environmentName -EnvironmentSpecification $environmentName -UserName $adminUserName -Password $adminPassword -Connection $connection -TaskContext $distributedTaskContext -ResourceFilter $machineFilter
-    Write-Verbose "Completed Register-Environment cmdlet call for environment : $environmentName"
-
-    $fetchedEnvironmentName = $environment.Name
-
-    Write-Verbose "Starting Get-EnvironmentResources cmdlet call on environment name: $fetchedEnvironmentName"
-    $resources = Get-EnvironmentResources -Environment $environment
-    Write-Verbose "Completed Get-EnvironmentResources cmdlet call for environment name: $fetchedEnvironmentName"
-
-    if ($resources.Count -eq 0)
+    if([string]::IsNullOrWhiteSpace($machineNames))
     {
-         throw (Get-LocalizedString -Key "No machine exists under environment: '{0}' for deployment" -ArgumentList $environmentName)
+        Write-Verbose "No environment found. Copying to destination."
```

Comment:
No *Machines* found....

Reply:
What is this scenario where we copy to local machine when machine name is empty. I thought we will make machine names a mandatory parameter.